### PR TITLE
Allow filtering of combined coverage reports

### PR DESCRIPTION
**Pull request by atomb**
Add a new `--only-label` option to `merge-coverage-reports` to include only one category of highlighting in the output. For example, merging coverage reports from test generation and verification using the option `--only-label NotCovered` will highlight only the regions not covered by either testing or verification.

### Description

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: Source/DafnyDriver/CoverageReporter.cs [hunk 1]
```diff
@@ -85,11 +85,15 @@ public class CoverageReporter {
...
+
+    var onlyLabel = options.Get(CoverageReportCommand.OnlyLabelOption);
     foreach (var report in reports) {
       foreach (var fileName in report.AllFiles()) {
         foreach (var span in report.CoverageSpansForFile(fileName)) {
           mergedReport.RegisterFile(span.Span.Uri);
-          mergedReport.LabelCode(span.Span, span.Label);
+          if ((onlyLabel ?? span.Label) == span.Label) {
+            mergedReport.LabelCode(span.Span, span.Label);
+          }
```
Comment (Dargones) [inline Source/DafnyDriver/CoverageReporter.cs:96]: Is the intent to display locations where both reports have the same label (i.e. intersection)? If yes, then there should be an else branch here that labels the code with the `NotApplicable` label. The report merged using the present approach will display the union between the two original reports.

**Review by Dargones — CHANGES_REQUESTED** [review]

File: Source/DafnyDriver/CoverageReporter.cs (hunk 1 above)
Comment (atomb) [reply Source/DafnyDriver/CoverageReporter.cs:96]: Not quite. The point is to highlight only the locations that, after merging (with union semantics), have the label specified on the command line. The expected output I've checked in looks the way I'd expect it to. The key use case I want to support is for `NotCovered`: highlight all locations that are not covered by either verification or tests.

Comment (atomb) [reply Source/DafnyDriver/CoverageReporter.cs:96]: Alternatively, you could see this code as taking the merged report and _removing_ all highlighting that isn't in the specified category. The conditional is arguably a bit confusing, given that it both checks whether the option has been specified and whether the current label matches the requested label in one expression.

**Review by Dargones — APPROVED** [review]

File: Source/DafnyDriver/CoverageReporter.cs (hunk 1 above)
Comment (Dargones) [reply Source/DafnyDriver/CoverageReporter.cs:96]: Ah, I see! So you can get the intersection if you don't use `--only-label` and the union if you do. This makes sense!

**Pull request merged**